### ISSUE TITLE
Add utility app to keep GPS on

### ISSUE THREAD

**Issue by andrixnet** [cgeo/cgeo#7557]
While using c:geo, if screen is turned off or focus is switched to another app, GPS is turned off. 
Reopening screen / refocusing on c:geo turns it on again, with the accompanying delay until fix is reestablished and further delays until the fix quality becomes optimal. 

This has been increasingly a problem on newer Android versions (since 4), as the system takes more aggressive measures to save battery power by turning off modules such as the GPS.

This has been discussed in 
https://github.com/cgeo/cgeo/issues/6549
https://github.com/cgeo/cgeo/issues/6649
https://github.com/cgeo/cgeo/issues/2262

In conclusion, c:geo does not provide the necessary service to keep GPS on in the given scenarios, but there are third party apps that provide this service, including dedicated, slim apps with only this function. 
There also are more complex apps that include track recording feature also provide this function indirectly.

This is a feature request that c:geo, under the section "utility programs" provide a recommendation for such an app as a helping hand towards users.

Thank you.

**Comment by jonas-koeritz** [comment]
This shouldn't be too hard to implement in c:geo itself. My proposal would be to add an option to the menu and when activated show an ongoing notification. The notification should be fitted with an action to disable the keepalive instantly to not confuse users.

**Comment by Bananeweizen** [comment]
I would not want this to be part of cgeo itself. That's something absolutely not essential to geocaching as such, but probably much more than just some lines of code to be done _right_ for a multitude of devices. If functionality is available in other apps and can be used immediately without any interference on cgeo, then we should really avoid creating more technical debt by adding the feature to our app. Otherwise we have to maintain it afterwards, and resources have been a long problem.
I'm therefore in favor of only adding this to the utilities menu. Users which don't find it there would also not find/use any such functionality being baked directly into cgeo.

**Comment by jonas-koeritz** [comment]
You're absolutely right, the amount of technical debt should not be underestimated.

**Comment by MagpieFourtyTwo** [comment]
> This is a feature request that c:geo, under the section "utility programs" provide a recommendation for such an app as a helping hand towards users.

Ok, gets a little bit OffTopic (not really an issue any more), but perhaps we can nevertheless sort it out here:

Which one(s) would you recommend?

I'm currently using "GPS Status & Toolbox" which comes with appropriate options too, as I've seen right now, like "keep GPS on when screen is switched off" (while switching it seems to make no difference at all, at least with c:geo ;) ) and seemingly even a floating widget (although I have not seen it yet ;) ). Moreover there seem to be problems with the pro key and the developer does not answer anymore (following some reviews in the store) - thus I would be interested in "better stuff".  ;)

**Comment by rsudev** [comment]
I personally use OruxMaps to track my path when doing longer caching tours. This keeps the GPS running and is not too heavy on battery. 
I would be interested in a smaller tool for the same purpose as well

**Comment by MagpieFourtyTwo** [comment]
Ah, yes, tracking: I sometimes use [Track My Trip](https://play.google.com/store/apps/details?id=com.neocor6.android.tmt&hl=de), but I never even thought of using it for keeping GPS on ... but it works. q. e. d. ;) And it's pretty lightweight. But have not yet tested how it performs in terms of battery consumption ...

**Comment by Lineflyer** [comment]
At the times devices needed a long time for GPS fix I once used the tool "GPS Locker" but AFAIR it had avertisement but it worked and once started it minimizes to a tray symbol.

**Comment by alan666notb** [comment]
I use also use "GPS Locker" [ https://play.google.com/store/apps/details?id=com.silentlexx.gpslock ], works fine, haven't seen any adverts.

**Comment by Lineflyer** [comment]
@alan666notb  
Yes, that is the app I also meant in my posting above. However it has advertisements unless you buy the premium version for 2,99€.
IMHO we can still add it. We also have GPS Status & Toolbox in there, which AFAIR also has ads in the basic version.

**Comment by alan666notb** [comment]
Ah yes - there is an advert (for Google pay on this occasion) when you start it up.
It's very unobtrusive - I wouldn't have known it was there but for you mentioning it.
I've never seen any pop-up ads or anything once it's running, it keeps a persistent notification in the status bar which is necessary to prevent android from shutting it down in the background.

**Comment by moving-bits** [comment]
I'll ask the developer if we can include their logo in our list.
